src/YoutubeAtom.tsx

```diff
@@ -197,121 +197,138 @@ export const YoutubeAtom = ({
+                                        eventEmitter('75'),
+                                    );
+                                    hasSent75Event = true;
+                                }
+
+                                const currentPlayerState =
+                                    player.current &&
+                                    (await player.current.getPlayerState());
+
+                                if (
+                                    currentPlayerState !==
+                                    youtubePlayerState.ENDED
+                                ) {
+                                    // Set a timeout to check progress again in the future
+                                    window.setTimeout(
+                                        () => checkProgress(),
+                                        3000,
+                                    );
+                                }
+                            };
+                        }
 
-                            if (
-                                currentPlayerState !== youtubePlayerState.ENDED
-                            ) {
-                                // Set a timeout to check progress again in the future
-                                window.setTimeout(() => checkProgress(), 3000);
-                            }
-                        };
-                    }
-
-                    if (e.data === youtubePlayerState.ENDED) {
-                        eventEmitters.forEach((eventEmitter) =>
-                            eventEmitter('end'),
-                        );
-                    }
-                },
-            );
+                        if (e.data === youtubePlayerState.ENDED) {
+                            eventEmitters.forEach((eventEmitter) =>
+                                eventEmitter('end'),
+                            );
+                        }
+                    },
+                );
 
-        return () => {
-            listener && player.current && player.current.off(listener);
-        };
-    }, [eventEmitters]);
+            return () => {
+                listener && player.current && player.current.off(listener);
+            };
+        }
+    }, [eventEmitters, iframeSrc]);
 
     return (
         <MaintainAspectRatio height={height} width={width}>
-            <iframe
-                title={title}
-                width={width}
-                height={height}
-                id={`youtube-video-${assetId}`}
-                src={iframeSrc}
-                // needed in order to allow `player.playVideo();` to be able to run
-                // https://stackoverflow.com/a/53298579/7378674
-                allow="autoplay"
-                tabIndex={overrideImage || posterImage ? -1 : 0}
-                allowFullScreen
-                data-atom-id={`youtube-video-${assetId}`}
-                data-atom-type="youtube"
-            />
+            {iframeSrc && (
```

Comment:
Maybe a better pattern would be to use `if (!iframeSrc) return null;` to bail out of the render